c_interface/c_interface.cpp

```diff
@@ -0,0 +1,118 @@
+#include "../inst/include/frogger.hpp"
+#include "../inst/include/generated/tensor_types.hpp"
+#include "../inst/include/generated/c_interface/c_types.hpp"
+#include "../inst/include/generated/c_interface/c_adapters.hpp"
+
+#include <windows.h>
+#include <numeric>
+#include <stdexcept>
+
+#define DllExport extern "C"
+#define EXPORT comment(linker, "/EXPORT:" __FUNCTION__ "=" __FUNCDNAME__)
+
+typedef void (WINAPI *CallbackFunction)(const char*);
+
+template <typename ModelVariant>
+HRESULT fit_model(leapfrog::internal::CParams<double> &data,
+                  leapfrog::internal::COptions &options,
+                  leapfrog::internal::COutput<double> &out,
+                  CallbackFunction error_handler) {
+
+  using LF = leapfrog::Leapfrog<leapfrog::C, double, ModelVariant>;
+
+  try {
+    std::vector<int> output_years((options.proj_end_year - options.proj_start_year) + 1);
+    std::iota(output_years.begin(), output_years.end(), options.proj_start_year);
+
+    const leapfrog::Options<double> opts =  {
+      10,
+      34,
+      leapfrog::internal::BaseSS::PROJPERIOD_CALENDAR,
+      options.proj_start_year,
+      options.proj_end_year
+    };
+    const auto pars = LF::Cfg::get_pars(data, opts);
+
+    auto state = LF::run_model(pars, opts, output_years);
+    LF::Cfg::build_output(0, state, out);
+  } catch (const std::invalid_argument& e) {
+    error_handler(e.what());
+    return E_INVALIDARG;
+  } catch (...) {
+    error_handler("Caught unhandled exception");
+    return E_FAIL;
+  }
```

Comment:
We should expand this for other potential error types, need to review specific errors we throw

Reply:
Added an issue #186